### ISSUE TITLE
Bad torrent poster image selected

### ISSUE THREAD

**Issue by feross** [feross/webtorrent-desktop#341]
File from archive.org: https://archive.org/download/AphexTwinBOACAphex/AphexTwinBOACAphex_archive.torrent

The torrent poster appears to be a zero-byte .jpg image. Let's not use a zero-byte image.

![screen shot 2016-04-06 at 21 40 48](https://cloud.githubusercontent.com/assets/121766/14340822/3c32a1da-fc40-11e5-9a5e-86e0aeb94e41.png)

**Comment by feross** [comment]
In other words, it's better if the default blue gradient background is used in this case.

**Comment by gdelavald** [comment]
Hello,
first contribution here so let me know if I can do anything differently.

I added a quick call to fs.syncStat on the poster to check the filesize after it gets saved, if it isn't zero-sized I save it as posterURL.

<img width="604" alt="screen shot 2016-04-07 at 2 13 36 pm" src="https://cloud.githubusercontent.com/assets/6303966/14360126/3bc56b3e-fccb-11e5-85b4-73e8243bce70.png">

Hopefully this is what was expected.

**Comment by AgentEpsilon** [comment]
@gdelavald Since `torrentPoster` creates a Buffer, wouldn't it be faster to just check `buf.length == 0`? Of course, I'm assuming that a 0 byte file would have a 0 byte buffer. If not, then your solution would be the only way. I'm checking that out now.

**Comment by AgentEpsilon** [comment]
Looks like the buffer size is indeed 0! @gdelavald, where/how did you find the default blue gradient image? I can't find it anywhere. Let me know, and I'll spin up a PR for this issue.

**Comment by gdelavald** [comment]
@AgentEpsilon in the function torrentPosterSaved it updates torrentSummary.posterURL as the posterPath, so if you don't update, when the function renderTorrent calls for the poster path it doesn't find the file and can't add it to the background image, leaving it as the default blue gradient.

This solution feels like exploiting a bug...

**Comment by AgentEpsilon** [comment]
@gdelavald Nah, when it executes the poster-related functions, they change the current poster from the default. Therefore, if those functions fail, the current poster never gets changed, leaving it as the default.

TL;DR It's not a bug, it's a feature.

**Comment by AgentEpsilon** [comment]
Sorry it took so long, still kinda a git newb :smiley:

**Comment by feross** [comment]
Thanks for working together to fix this @gdelavald and @AgentEpsilon. Nice work!
